akka-stream/src/main/scala/akka/stream/impl/io/TcpListenStreamActor.scala

```diff
@@ -151,10 +155,17 @@ private[akka] class TcpListenStreamActor(localAddressPromise: Promise[InetSocket
     val (connected: Connected, connection: ActorRef) = incomingConnections.dequeueInputElement()
     val tcpStreamActor = context.watch(context.actorOf(TcpStreamActor.inboundProps(connection, halfClose, settings)))
     val processor = ActorProcessor[ByteString, ByteString](tcpStreamActor)
+
+    import scala.concurrent.duration._
+    val handler = (idleTimeout match {
+      case d: FiniteDuration ⇒ Flow[ByteString].join(Timeouts.idleTimeoutBidi[ByteString, ByteString](d))
+      case _                 ⇒ Flow[ByteString]
+    }).andThenMat(() ⇒ (processor, ()))
```

Comment:
Hm... this *is* tcp handling internals though, there is stuff like that on these levels - i.e. materializing the connection stream twice will not work, and we tell people about this.